usim/_basics/tracked.py

```diff
@@ -129,94 +99,85 @@ def value(self) -> V:
-            await listener.__on_changed__()
+        for listener in list(self._listeners):
+            listener.__on_changed__()
         await postpone()
 
-    async def __set_expression__(self, to: V):
-        await self.set(to)
-        return self
-
-    def _async_operation(
-            self, op: Callable[[V, RHS], V], rhs: RHS,
-    ) -> 'TrackedOperation[V]':
-        return TrackedOperation(self, Operation(op, rhs))
-
-    # boolean operations producing a BoolExpression
+    # boolean operations producing a AsyncComparison
     def __lt__(self, other):
-        return BoolExpression(operator.lt, self, other)
+        return AsyncComparison(operator.lt, self, other)
 
     def __le__(self, other):
-        return BoolExpression(operator.le, self, other)
+        return AsyncComparison(operator.le, self, other)
 
     def __eq__(self, other):
-        return BoolExpression(operator.eq, self, other)
+        return AsyncComparison(operator.eq, self, other)
 
     def __ne__(self, other):
-        return BoolExpression(operator.ne, self, other)
+        return AsyncComparison(operator.ne, self, other)
 
     def __ge__(self, other):
-        return BoolExpression(operator.gt, self, other)
+        return AsyncComparison(operator.ge, self, other)
 
     def __gt__(self, other):
-        return BoolExpression(operator.ge, self, other)
+        return AsyncComparison(operator.gt, self, other)
 
     # modifying operators
-    def __add__(self, other):
-        return self.__set_expression__(self.value + other)
+    def __add__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__add__, other)
 
-    def __sub__(self, other):
-        return self.__set_expression__(self.value - other)
+    def __sub__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__sub__, other)
 
-    def __mul__(self, other):
-        return self.__set_expression__(self.value * other)
+    def __mul__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__mul__, other)
 
-    def __matmul__(self, other):
-        return self.__set_expression__(self.value @ other)
+    def __matmul__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__matmul__, other)
 
-    def __truediv__(self, other):
-        return self.__set_expression__(self.value / other)
+    def __truediv__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__truediv__, other)
 
-    def __floordiv__(self, other):
-        return self.__set_expression__(self.value // other)
+    def __floordiv__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__floordiv__, other)
 
-    def __mod__(self, other):
-        return self.__set_expression__(self.value % other)
+    def __mod__(self, other) -> 'AsyncOperation[V]':
+        return AsyncOperation(self, operator.__mod__, other)
 
-    def __divmod__(self, other):
```

Comment:
No replacement for `__divmod__`?

Reply:
This is intentional - writing the unittests revealed that treating [``__divmod__``](https://docs.python.org/3/library/functions.html#divmod) like a regular operator, say ``__add__``, makes no sense. :D

Regular operators have the signature ``(T, T) -> T``, but ``divmod`` has the signature ``(T, T) -> (T, T)``. This makes it unsuitable to update the wrapped value, since it is not clear which result to use.